fairlearn/postprocessing/plotting.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
```

Comment:
Maybe this was discussed somewhere. Why not just have a method `plot` on ThresholdOptimizer?

Reply:
@adrinjalali  recommended following the same pattern that users might be used to from sklearn. I don't think it's better one way or the other, but if that's what users are used to then that's how it should be. What do you think @MiroDudik ?

Reply:
Yeah, `plot` is not what people expect to see as a method on an estimator

Reply:
I can see that. I don't have a strong preference, so happy to go with the standalone function.

However, I would still suggest a minor refactoring:
* create just the function `plot_threshold_optimizer` since the plotting is highly specific to plotting the solution of a *fitted* threshold optimizer; and only one kind of a plot applies to any instance anyway.
* don't put `plot_threshold_optimizer` into a standalone module, but instead put it in `fairlearn.postprocessing`. I think it's better to have `ThresholdOptimizer` and `plot_threshold_optimizer` in the same module (in our case, `fairlearn.postprocessing`). This is similar to `roc_curve` and `plot_roc_curve` being both in `sklearn.metrics`.

Reply:
I can't put it under fairlearn.postprocessing because that doesn't necessarily guarantee the installation of matplotlib.

I like your rationale for a `plot_threshold_optimizer` method, although I might adjust the name slightly.

Reply:
it's a bit awkward to have modules with singleton functions. can we try something like this?
https://github.com/scikit-learn/scikit-learn/blob/b194674c42d54b26137a456c510c5fdba1ba23e0/sklearn/metrics/_plot/roc_curve.py#L84

Reply:
Adjusted to `plot_threshold_optimizer`, but still in a separate module. @MiroDudik if you really want that we need to add the matplotlib dependency back

Reply:
Agreed that we shouldn't require matplotlib to always install. In sklearn, matplotlib is not automatically installed, but `plot_*` might then throw `ImportError`. See:
[sklearn dependencies](https://pypi.org/project/scikit-learn/)
[how ImportError is invoked](https://github.com/scikit-learn/scikit-learn/blob/d4a3ac4252401a20368c405976ff3db6f30995ac/sklearn/utils/__init__.py#L1094)
Is that too much hassle?

Reply:
Done!